client/src/components/FormBuilderModal/FormBuilderModal.js

```diff
@@ -1,104 +1,71 @@
 import castStringToElement from 'lib/castStringToElement';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
+import Modal from 'components/Modal/Modal';
 
 const noop = () => null;
 
-class FormBuilderModal extends Component {
-  constructor(props) {
-    super(props);
-
-    this.handleSubmit = this.handleSubmit.bind(this);
-    this.handleHide = this.handleHide.bind(this);
-    this.clearResponse = this.clearResponse.bind(this);
-    this.handleLoadingError = this.handleLoadingError.bind(this);
-  }
-
-  /**
-   * Defines the form part of the Modal
-   *
-   * @returns {Component}
-   */
-  getForm() {
-    const { FormBuilderLoaderComponent } = this.props;
-    if (!this.props.schemaUrl) {
-      return null;
-    }
-    return (
-      <FormBuilderLoaderComponent
-        fieldHolder={{ className: classnames('modal-body', this.props.bodyClassName) }}
-        actionHolder={{ className: 'modal-footer' }}
-        autoFocus={this.props.autoFocus}
-        schemaUrl={this.props.schemaUrl}
-        onSubmit={this.handleSubmit}
-        onAction={this.props.onAction}
-        onLoadingError={this.handleLoadingError}
-        identifier={this.props.identifier}
-      />
-    );
-  }
-
-  /**
-   * Generates the response part of the Modal
-   *
-   * @returns {Component}
-   */
-  getResponse() {
-    if (!this.state || !this.state.response) {
-      return null;
-    }
-
-    let className = '';
-
-    if (this.state.error) {
-      className = this.props.responseClassBad;
-    } else {
-      className = this.props.responseClassGood;
-    }
-
-    return (
-      <div className={className}>
-        { castStringToElement('span', { html: this.state.response }) }
-      </div>
-    );
-  }
-
-  /**
-   * Removes the response from the state
-   */
-  clearResponse() {
-    this.setState({
-      response: null,
-    });
-  }
-
-  handleLoadingError(schema) {
-    const providesOnLoadingError = this.props.onLoadingError !== noop;
-    if (this.props.showErrorMessage || !providesOnLoadingError) {
-      const error = schema.errors && schema.errors[0];
-      this.setState({
-        response: error.value,
-        error: true,
-      });
+/**
+ * React component for displaying a Form in a Modal using a form schema URL
+ */
+const FormBuilderModal = ({
+  children,
+  FormBuilderLoaderComponent,
+  onLoadingError,
+  onSubmit,
+  responseClassBad,
+  responseClassGood,
+  showErrorMessage,
+
+  // Form builder props
+  autoFocus,
+  bodyClassName,
+  identifier,
+  onAction,
+  schemaUrl,
+
+  // Props pass to modal
+  className,
+  isOpen,
+  modalClassName,
+  ModalComponent,
+  ModalHeaderComponent,
+  onClosed,
+  showCloseButton,
+  size,
+  title,
+}) => {
+  /** @var {string} response Message we got back from posting the form. */
+  const [response, setResponse] = useState(null);
+  /** @var {boolean} response Whether the response was an error or not. */
+  const [error, setError] = useState(null);
```

Comment:
```suggestion
  const [response, setResponse] = useState(null);
  const [error, setError] = useState(null);
```
We're not use `@var` elsewhere in the codebase

Reply:
It's a standard JSDoc feature which we are using in plenty of other context. 

It's useful for type hitting and most IDEs will pick it up.